ts/Series/Heatmap/HeatmapSeries.ts

```diff
@@ -449,14 +449,19 @@ class HeatmapSeries extends ScatterSeries {
             if (canvas && ctx && colorAxis) {
                 const width = canvas.width - 1,
                     height = canvas.height - 1,
-                    { image, data, scaleValue } = heatmap,
+                    data = heatmap.data,
+                    image = heatmap.image,
                     { min: xMin, max: xMax } = heatmap.getXExtremes(
                         heatmap.xData || []
                     ),
                     { dataMin: yMin, dataMax: yMax } = heatmap.getExtremes(),
                     xSpan = xMax - xMin,
                     ySpan = (yMax as number) - (yMin as number);
 
+                if (image) {
+                    image.destroy();
```

Comment:
Are you sure? Is it efficient to destroy and rebuild every time the chart redraws?

Reply:
The thing is, I was a bit unsure if there is another way to update the image in case of new data. 

I guess destroying it on every redraw is a bit much, but do we have some internal property perhaps that tells us data has changed (thus an image needs a new source)?

Or can we just assign a new image to the heatmap-series, and the previous one will go out of scope?

Reply:
I don't think this code executes at all if the data doesn't require updating. What I think is most efficient is to create the image only once, resize it only if a size change is needed (compare the existing width/height against the required width/height), and modify the pixels, which is probably always necessary at this point.

Reply:
Turns out it was not necessary to do anything with the content of the image - it reflects new data via the update-function, and as far as I can tell it does not create a new image. I also made it so it only resizes when needed as you describe.